parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -325,13 +326,17 @@ def get_version_lister(self):
         finally:
             await self.transactions.fd_close(fd)
 
-    async def write_bytes(self, path: AnyPath, data: bytes, offset: int = 0) -> int:
+    async def write_bytes(
+        self, path: AnyPath, data: bytes, offset: int = 0, truncate: bool = True
+    ) -> int:
         """
         Raises:
             FSError
         """
         path = FsPath(path)
         _, fd = await self.transactions.file_open(path, "w")
+        if offset >= 0 and truncate:
+            await self.transactions.fd_resize(fd, offset)
```

Comment:
This should go inside the `try-finally` block in order to make sure that the file descriptor gets closed.